aws_lambda_powertools/utilities/data_classes/s3_batch_operation_event.py

```diff
@@ -0,0 +1,237 @@
+
+    ```
+    """  # noqa: E501
+
+    invocation_schema_version: str
+    invocation_id: str
+    treat_missing_keys_as: Literal["Succeeded", "TemporaryFailure", "PermanentFailure"] = "Succeeded"
+    results: List[S3BatchOperationResponseRecord] = field(default_factory=list)
+
+    def __post_init__(self):
+        if self.treat_missing_keys_as not in VALID_RESULT_CODE_TYPES:
+            warnings.warn(
+                stacklevel=2,
+                message=f"The value {self.treat_missing_keys_as} is not valid for treat_missing_keys_as, "
+                "Choose from 'Succeeded', 'TemporaryFailure', 'PermanentFailure'",
+            )
+
+    def add_result(self, result: S3BatchOperationResponseRecord):
+        self.results.append(result)
+
+    def asdict(self) -> Dict:
+        result_count = len(self.results)
+
+        if result_count != 1:
+            raise ValueError(f"Response must have exactly one result, but got {result_count}")
+
+        return {
+            "invocationSchemaVersion": self.invocation_schema_version,
+            "treatMissingKeysAs": self.treat_missing_keys_as,
+            "invocationId": self.invocation_id,
+            "results": [result.asdict() for result in self.results],
+        }
+
+
+class S3BatchOperationJob(DictWrapper):
+    @property
+    def get_id(self) -> str:
+        # Note: this name conflicts with existing python builtins
+        return self["id"]
+
+    @property
+    def user_arguments(self) -> Optional[Dict[str, str]]:
+        """Get user arguments provided for this job (only for invocation schema 2.0)"""
+        return self.get("userArguments")
+
+
+class S3BatchOperationTask(DictWrapper):
+    @property
+    def task_id(self) -> str:
+        """Get the task id"""
+        return self["taskId"]
+
+    @property
+    def s3_key(self) -> str:
+        """Get the object key using unquote_plus"""
+        return unquote_plus(self["s3Key"])
+
+    @property
+    def s3_version_id(self) -> Optional[str]:
+        """Object version if bucket is versioning-enabled, otherwise null"""
+        return self.get("s3VersionId")
+
+    @property
+    def s3_bucket_arn(self) -> Optional[str]:
+        """Get the s3 bucket arn (present only for invocationSchemaVersion '1.0')"""
+        return self.get("s3BucketArn")
+
+    @property
+    def s3_bucket(self) -> str:
+        """ "
+        Get the s3 bucket, either from 's3Bucket' property (invocationSchemaVersion '2.0')
+        or from 's3BucketArn' (invocationSchemaVersion '1.0')
+        """
+        if self.s3_bucket_arn:
+            return self.s3_bucket_arn.split(":::")[-1]
+        return self["s3Bucket"]
+
+    def build_task_batch_response(
+        self,
+        result_code: Literal["Succeeded", "TemporaryFailure", "PermanentFailure"] = "Succeeded",
+        result_string: str = "",
+    ) -> S3BatchOperationResponseRecord:
+        """Create a S3BatchOperationResponseRecord directly using the record_id and given values
```

Comment:
```suggestion
        """Create a S3BatchOperationResponseRecord directly using the recordId and given values
```

Reply:
Reading the code, I don't understand where the `record_id` mentioned here is coming from. Can you explain?

Reply:
Yeah, it was a mistake, resolved.